homekit/StateManager.js

```diff
@@ -1,623 +1,1039 @@
 
+			/** @param {homebridge.CharacteristicGetCallback} callback */
 			StatusLowBattery: (callback) => {
+				if (!(device.state instanceof Classes.InternalSensorState)) {
+					// TODO: log warning
+					return
+				}
+
 				const lowBattery = device.state.lowBattery
 
-				log.easyDebug(device.name, '(GET) - Status Low Battery:', lowBattery)
+				easyDebugInfo(device.name, '(GET) - Status Low Battery:', lowBattery)
 
-				callback(null, Characteristic.StatusLowBattery[lowBattery])
+				callback(null, this.Characteristic.StatusLowBattery[lowBattery])
 			},
 
 			// HORIZONTAL SWING
+			/** @param {homebridge.CharacteristicGetCallback} callback */
 			HorizontalSwing: (callback) => {
+				if (!(device.state instanceof Classes.InternalAcState)) {
+					// TODO: log warning
+					return
+				}
+
 				const horizontalSwing = device.state.horizontalSwing
 
-				log.easyDebug(device.name, '(GET) - Horizontal Swing:', horizontalSwing)
+				easyDebugInfo(device.name, '(GET) - Horizontal Swing:', horizontalSwing)
 
 				callback(null, horizontalSwing === 'SWING_ENABLED')
 			},
 
 			// AIR CONDITIONER/PURIFIER LIGHT
+			/** @param {homebridge.CharacteristicGetCallback} callback */
 			LightSwitch: (callback) => {
+				if (!(device.state instanceof Classes.InternalAcState || device.state instanceof Classes.InternalAirPurifierState)) {
+					// TODO: log warning
+					return
+				}
+
 				const light = device.state.light
 
-				log.easyDebug(device.name, '(GET) - Light:', light ? 'ON' : 'OFF')
+				easyDebugInfo(device.name, '(GET) - Light:', light ? 'ON' : 'OFF')
 
 				callback(null, light)
 			},
 
 			// CLIMATE REACT
+			/** @param {homebridge.CharacteristicGetCallback} callback */
 			ClimateReactSwitch: (callback) => {
+				if (!(device.state instanceof Classes.InternalAcState)) {
+					// TODO: log warning
+					return
+				}
+
 				const smartModeEnabled = device.state.smartMode.enabled
 
-				log.easyDebug(device.name, '(GET) - Climate React Enabled Switch:', smartModeEnabled)
+				easyDebugInfo(device.name, '(GET) - Climate React Enabled Switch:', smartModeEnabled)
 
 				callback(null, smartModeEnabled)
 			},
 
 			// OCCUPANCY SENSOR
+			/** @param {homebridge.CharacteristicGetCallback} callback */
 			OccupancyDetected: (callback) => {
+				if (!(device.state instanceof Classes.InternalOccupancyState)) {
+					// TODO: log warning
+					return
+				}
+
 				const occupancy = device.state.occupancy
 
-				log.easyDebug(device.name, '(GET) Occupancy Detected:', occupancy)
+				easyDebugInfo(device.name, '(GET) Occupancy Detected:', occupancy)
 
-				callback(null, Characteristic.OccupancyDetected[occupancy])
+				callback(null, this.Characteristic.OccupancyDetected[occupancy])
 			},
 
 			// Air Quality
+			/** @param {homebridge.CharacteristicGetCallback} callback */
 			AirQuality: (callback) => {
+				if (!(device.state instanceof Classes.InternalAirPurifierState || device.state instanceof Classes.InternalAirQualitySensorState)) {
```

Comment:
Based on other changes I think this can be limited to only InternalAirQualitySensorState
```suggestion
				if (!(device.state instanceof Classes.InternalAirQualitySensorState)) {
```